nav2_bt_navigator/src/bt_navigator.cpp

```diff
@@ -116,6 +117,9 @@ BtNavigator::on_configure(const rclcpp_lifecycle::State & /*state*/)
   blackboard->set<bool>("initial_pose_received", false);  // NOLINT
   blackboard->set<int>("number_recoveries", 0);  // NOLINT
 
+  // Odometry smoother object for getting current speed
+  odom_smoother_ = std::make_unique<nav2_util::OdomSmoother>(rclcpp_node_, 0.3);
```

Comment:
Probably a better idea to pass the `bt_navigator` node shared_ptr using `shared_from_this()`. I can't see an `rclcpp_node_` object defined anywhere.

Reply:
I have tried that and it doesn't work as the `OdomSmoother` class only accepts a reference to  `rclcpp::Node::WeakPtr` and not a `LifecycleNode`. The `rclcpp_node_` object is defined and initialized in the parent class of `BtNavigator` (`nav2_util::LifecycleNode`) here: https://github.com/ros-planning/navigation2/blob/3bf1f7a93d740c54b4c7ef7fc6dd914175bf46be/nav2_util/src/lifecycle_node.cpp#L62. And since it is a protected member, it should be available for this object to access. And as stated in the header file here https://github.com/ros-planning/navigation2/blob/3bf1f7a93d740c54b4c7ef7fc6dd914175bf46be/nav2_util/include/nav2_util/lifecycle_node.hpp#L194 that seems to be the intent behind this member variable.